### DISCUSSION TITLE
How do I prevent VAAPI from filling up gigabytes worth of space in container?

### DISCUSSION THREAD

**Discussion by rfrantzen** [blakeblackshear/frigate#13729] [Ask A Question] [answered]
Is there a way to prevent what seems like VAAPI from writing all of theses logs in /opt/frigate within the container? It's consumes gigabytes per day and I have to keep deleting them:

The files in /opt/frigate look like this:

36048 -rw-r--r-- 1 root root 36907499 Sep 13 18:35 0.222611.thd-0x000002a2
36036 -rw-r--r-- 1 root root 36896286 Sep 13 18:35 0.222611.thd-0x000002a3
 9592 -rw-r--r-- 1 root root  9816329 Sep 13 18:35 0.222612.thd-0x0000024b
 4796 -rw-r--r-- 1 root root  4904599 Sep 13 18:35 0.222612.thd-0x000002ae
 4752 -rw-r--r-- 1 root root  4861290 Sep 13 18:35 0.222612.thd-0x000002af
15340 -rw-r--r-- 1 root root 15701703 Sep 13 18:51 0.223645.thd-0x00000234
57160 -rw-r--r-- 1 root root 58525023 Sep 13 18:51 0.223645.thd-0x00000258
57136 -rw-r--r-- 1 root root 58499671 Sep 13 18:51 0.223645.thd-0x00000259
15328 -rw-r--r-- 1 root root 15687991 Sep 13 18:51 0.223646.thd-0x00000216
15320 -rw-r--r-- 1 root root 15681135 Sep 13 18:51 0.223646.thd-0x0000023d
56632 -rw-r--r-- 1 root root 57986395 Sep 13 18:51 0.223646.thd-0x0000025e
56644 -rw-r--r-- 1 root root 57996710 Sep 13 18:51 0.223646.thd-0x0000025f
56640 -rw-r--r-- 1 root root 57991782 Sep 13 18:51 0.223646.thd-0x0000026c
56640 -rw-r--r-- 1 root root 57993780 Sep 13 18:51 0.223646.thd-0x0000026d
15260 -rw-r--r-- 1 root root 15619431 Sep 13 18:51 0.223648.thd-0x00000204
15316 -rw-r--r-- 1 root root 15676568 Sep 13 18:51 0.223648.thd-0x00000217
15244 -rw-r--r-- 1 root root 15604527 Sep 13 18:51 0.223648.thd-0x0000023a
56732 -rw-r--r-- 1 root root 58088995 Sep 13 18:51 0.223648.thd-0x00000281
56700 -rw-r--r-- 1 root root 58054716 Sep 13 18:51 0.223648.thd-0x00000282
38360 -rw-r--r-- 1 root root 39276192 Sep 13 18:51 0.223648.thd-0x00000283
38400 -rw-r--r-- 1 root root 39314355 Sep 13 18:51 0.223648.thd-0x00000284
 7592 -rw-r--r-- 1 root root  7771072 Sep 13 18:51 0.223648.thd-0x00000293
 7532 -rw-r--r-- 1 root root  7708644 Sep 13 18:51 0.223648.thd-0x00000295
15284 -rw-r--r-- 1 root root 15643427 Sep 13 18:51 0.223649.thd-0x0000020d
57304 -rw-r--r-- 1 root root 58674430 Sep 13 18:51 0.223649.thd-0x00000296
57308 -rw-r--r-- 1 root root 58676484 Sep 13 18:51 0.223649.thd-0x00000297
    8 drwxr-xr-x 1 root root     4096 Sep 12 16:52 frigate
    4 drwxr-xr-x 2 root root     4096 Aug 28 08:27 migrations
    4 drwxr-xr-x 5 root root     4096 Aug 28 08:29 web

The logs start like this:
[48768.899994][ctx       none]==========va_TraceInitialize
[48768.900016][ctx       none]=========vaInitialize ret = VA_STATUS_SUCCESS, success (no error) 
[48769.878651][ctx       none]==========va_TraceCreateConfig
[48769.878653][ctx       none]  profile = -1, VAProfileNone
[48769.878654][ctx       none]  entrypoint = 10, VAEntrypointVideoProc
[48769.878654][ctx       none]  num_attribs = 0
[48769.878694][ctx       none]=========vaCreateConfig ret = VA_STATUS_SUCCESS, success (no error) 
[48769.878698][ctx       none]==========va_TraceQuerySurfaceAttributes
[48769.878698][ctx       none]  config = 0x00000800
[48769.878708][ctx       none]=========vaQuerySurfaceAttributes ret = VA_STATUS_SUCCESS, success (no error) 
[48769.878723][ctx       none]==========va_TraceQuerySurfaceAttributes
[48769.878724][ctx       none]  config = 0x00000800
[48769.878724][ctx       none]  attrib_list[0] =
[48769.878724][ctx       none]          type = 1
[48769.878725][ctx       none]          flags = 3
[48769.878725][ctx       none]          value.type = 1
[48769.878725][ctx       none]          value.value.i = 0x3231564e
[48769.878726][ctx       none]  attrib_list[1] =
[48769.878726][ctx       none]          type = 3
[48769.878726][ctx       none]          flags = 3
[48769.878726][ctx       none]          value.type = 1
[48769.878727][ctx       none]          value.value.i = 0x00004000
[48769.878727][ctx       none]  attrib_list[2] =
[48769.878727][ctx       none]          type = 5
[48769.878727][ctx       none]          flags = 3
[48769.878728][ctx       none]          value.type = 1
[48769.878728][ctx       none]          value.value.i = 0x00004000

Setting the following environment variables doesn't stop them either:
LIBVA_LOG_LEVEL=0
LIBVA_MESSAGING_LEVEL=0

Is something in the Frigate Docker container forcing these logs? I'm running the latest version 0.14.1-f4f3cfa.

Relevant ffmpeg settings in config:

 # Required: ffmpeg settings for the camera
    ffmpeg:
      #output_args:
      #  record: preset-record-ubiquiti
      output_args:
        record: preset-record-generic-audio-aac
      inputs:
        # Required: the path to the stream
        # NOTE: path may include environment variables, which must begin with 'FRIGATE_' and be referenced in {}
        - path: rtsp://127.0.0.1:8554/doorbell_sub
          input_args: preset-rtsp-restream
          # Required: list of roles for this stream. valid values are: detect,record,rtmp
          # NOTICE: In addition to assigning the record and rtmp roles,
          # they must also be enabled in the camera config.
          roles:
            - detect
          
        - path: rtsp://127.0.0.1:8554/doorbell
          input_args: preset-rtsp-restream

**Comment by NickM-27** [comment]
I don't think this has anything to do with frigate, I use vaapi and have never seen this

**Reply by rfrantzen** [reply]
Removing the Frigate docker container immediately reclaims all of the space on my host, so there seems to be a correlation.

**Comment by rfrantzen** [comment]
If I run vainfo from within the Frigate container bash I see:

root@homeassistant:/opt/frigate# vainfo
Authorization required, but no authorization protocol specified
error: can't connect to X server!
libva info: Open new log file 0.235243.thd-0x00000368 for the thread 0x00000368
libva info: LIBVA_TRACE is on, save log into 0.235243.thd-0x00000368
libva info: VA-API version 1.17.0

It says LIBVA_TRACE is on, so that is likely my problem. Does it say trace is on or off for you? If off then I guess I need to see what's going on with my host. Thanks.

**Reply by NickM-27** [reply]
I don't see anything regarding LIBVA_TRACE

```
error: XDG_RUNTIME_DIR is invalid or not set in the environment.
error: can't connect to X server!
libva info: VA-API version 1.17.0
libva info: User environment variable requested driver 'radeonsi'
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/radeonsi_drv_video.so
libva info: Found init function __vaDriverInit_1_17
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.17 (libva 2.10.0)
vainfo: Driver version: Mesa Gallium driver 22.3.6 for AMD Radeon Graphics (renoir, LLVM 15.0.6, DRM 3.49, 6.1.106-Unraid)
vainfo: Supported profile and entrypoints
```

**Reply by rfrantzen** [reply, marked as answer]
I noticed that the docker was running in Privileged mode. Once I got rid of that the problem (from my host) was no longer spilling into the container. I was able to unset a couple LIBVA environment variables and that turned off the trace system-wide and all is good now. Thanks for confirming yours doesn't have that on.
